Sming/Wiring/FakePgmSpace.h

```diff
@@ -170,18 +147,13 @@ extern "C"
 
 #else /* ICACHE_FLASH */
 
-#define PROGMEM
-
-#define PSTR(_str) (_str)
-#define _F(_str) (_str)
+#define PROGMEM __attribute__((aligned(4)))
```

Comment:
This will break the compilation of some Arduino and other libraries that define PROGMEM as empty for platforms other than AVR or ESP8266. As long as redefinition is treated as error from the compiler. 
For example: https://github.com/SmingHub/Sming/blob/develop/Sming/Libraries/Adafruit_GFX/glcdfont.c#L10 has the following code:
```
#if defined(__AVR__)
 #include <avr/io.h>
 #include <avr/pgmspace.h>
#elif defined(__ESP8266_EX__)
 #include <WiringFrameworkDependencies.h>
#else
 #define PROGMEM
#endif
```